osu.Game/Screens/Select/Carousel/CarouselItem.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using System.Collections.Generic;
+using osu.Framework.Configuration;
+
+namespace osu.Game.Screens.Select.Carousel
+{
+    public abstract class CarouselItem
+    {
+        public readonly BindableBool Filtered = new BindableBool();
+
+        public readonly Bindable<CarouselItemState> State = new Bindable<CarouselItemState>(CarouselItemState.NotSelected);
+
+        public IReadOnlyList<CarouselItem> Children => InternalChildren;
+
+        protected List<CarouselItem> InternalChildren { get; set; }

```

Comment:
How about moving Children and InternalChildren to CarouselGroup? Would make more sense to me for groups to contain items, rather than items containing items.